src/server/pfs/server/driver.go

```diff
@@ -632,6 +633,45 @@ func (d *driver) isPathModifiedInCommit(ctx context.Context, commit *pfs.Commit,
 	return found, nil
 }
 
+func (d *driver) getCompactedDiffFileSet(ctx context.Context, commit *pfs.Commit) (*fileset.ID, error) {
+	var compactedDiff *fileset.ID
+	return compactedDiff, backoff.RetryUntilCancel(ctx, func() error {
+		diff, err := d.getDiffFileSetIfCompacted(ctx, commit)
+		if err != nil {
+			return err
+		}
+		if diff != nil {
+			compactedDiff = diff
+			return nil
+		}
+		return errors.Errorf("diff file set for commit %s is not compacted yet", commit.ID)
+	}, backoff.NewConstantBackOff(time.Second), backoff.NotifyCtx(ctx, "getCompactedDiffFileSet for "+commit.ID))
```

Comment:
We should discuss the backoff configuration more. Polling every second seems excessive.

Reply:
I think last time we discussed this, we acknowledged that 1 second was probably too aggressive generally, but it would lead to a better user experience from the Console perspective. Happy to adjust it to something less aggressive if we think its necessary.